spec/factories/service_providers.rb

```diff
@@ -18,5 +18,13 @@
     trait :team do
       association :team, factory: :team
     end
+
+    trait :with_users_team do
+      after(:build) do |service_provider|
+        team = service_provider.user&.teams[1] || create(:team)
```

Comment:
I'm not sure I follow this line, is the goal to create a second team if one does not exist already?

Reply:
Good point -- if the user doesn't have a team, the team will be nil and the test will (presumably) fail, which likely indicates a problem with the test set up.